### PR TITLE
Enhance RPM packages retrieving support

### PR DESCRIPTION
**Pull request by jnasselle**
|Related issue|
|---|
|#10324|

## Description

Hi team!

This PR aims to enhance RPM packages retrieving support by basically adding two new mechanisms to obtain such information:
- Using `librpm` 4.16.3 library: add support for RPM databases other than Berkley DB, i.e NDB and SQLITE, current formats for distros like openSUSE Tumbleweed and Fedora 34 or newer
- Executing `rpm -qa`: In case any of the 'formal' mechanisms to retrieve package information fails (via `librpm` or `libdb`), the `rpm` CLI tool will be used as a fallback provider.

Design considerations:
- Using stable `libdb` (Berkley DB support) to read Berkley DB rather than `librpm` experimental and his `read-only BDB`
- RPM database name will determine whether is BDB or other one, and the correct handling. The existence of `/var/lib/rpm/Packages` will indicate that libdb should be used, and librpm otherwise, allowing to parse `/var/lib/rpm/rpm.sqlite` for Fedora 34 or `/var/lib/rpm/Packages.db` for openSUSE Tumbleweed.
- Fallback mode (`rpm -qa`) in executed in case of exception of `libdb` or `librpm`
- Centos 5 will not use a fallback or librpm mechanism based on the criteria that is truly difficult (impossible) to have a scenario of Centos 5 and RPM>=4.16

**NOTES**
- This enhancement is not valid for CentOS 5
- librpm is also included in deb-based packages/systems.
### External deps source packages size
```
du -ch popt.tar.gz rpm.tar.gz
872K	popt.tar.gz
2.1M	rpm.tar.gz
3.0M	total
```
### External deps precompiled packages size
```
du -ch popt.tar.gz rpm.tar.gz
1.2M	popt.tar.gz
2.9M	rpm.tar.gz
4.1M	total
```
### RPM Packages
- RPM i386 agent
Current size: 8.1MB. [Reference package](https://packages-dev.wazuh.com/staging/yum/wazuh-agent-4.4.0-0.40400.20211202.i386.rpm)
New size:  8.5MB. [Reference package](https://packages-dev.wazuh.com/warehouse/pullrequests/4.4/rpm/var/wazuh-agent-4.4.0-0.commit33c3c43.i386.rpm)
- RPM x86_64 agent
Current size: 8.5MB. [Reference package](https://packages-dev.wazuh.com/staging/yum/wazuh-agent-4.4.0-0.40400.20211202.x86_64.rpm)
New size:  8.8MB. [Reference package](https://packages-dev.wazuh.com/warehouse/pullrequests/4.4/rpm/var/wazuh-agent-4.4.0-0.commit33c3c43.x86_64.rpm)

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] OpenSuse Tumbleweed. https://ci.wazuh.info/job/Test_install/79077/
  - [x] Fedora 35 
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [ ] Review logs syntax and correct language
- [x] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [x] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [x] The data flow works as expected (agent-manager-api-app)
- [x] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

**Review by Dwordcito — DISMISSED** [review]

File: src/data_provider/tests/sysInfoRpmPackageManager/RpmPackageManager_test.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * March 16, 2021.
```
Comment (HanesSciarrone) [inline src/data_provider/tests/sysInfoRpmPackageManager/RpmPackageManager_test.h:4]: Hi Team

This file is new, the date must be this month or last month.

Regards
Hanes

File: src/data_provider/tests/sysInfoRpmPackageManager/main.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * March 16, 2021.
```
Comment (HanesSciarrone) [inline src/data_provider/tests/sysInfoRpmPackageManager/main.cpp:4]: Hi Team

This file is new, the date must be this month or last month.

Regards
Hanes

File: src/data_provider/src/packages/packageLinuxParserRpmLegacy.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/* Copyright (C) 2015-2021, Wazuh Inc.
+ * All rights reserved.
+ *
```
Comment (HanesSciarrone) [inline src/data_provider/src/packages/packageLinuxParserRpmLegacy.cpp:3]: Hi Team

The creation date is missing.

Regards
Hanes

File: src/data_provider/src/packages/packageLinuxRpmParserHelper.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * January 28, 2021.
```
Comment (HanesSciarrone) [inline src/data_provider/src/packages/packageLinuxRpmParserHelper.h:4]: Hi Team

I think the date is wrong

Regards
Hanes

File: src/data_provider/src/packages/packageLinuxRpmParserHelperLegacy.h [hunk 5]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * January 28, 2021.
```
Comment (HanesSciarrone) [inline src/data_provider/src/packages/packageLinuxRpmParserHelperLegacy.h:4]: Hi Team

I think the date is wrong

Regards
Hanes

**Review by HanesSciarrone — CHANGES_REQUESTED** [review]

File: src/data_provider/src/packages/packageLinuxRpmParserHelper.h (hunk 4 above)
Comment (RamiroRD) [reply src/data_provider/src/packages/packageLinuxRpmParserHelper.h:4]: These are files that were copied and renamed from already existing files with this date. I'm not sure whether we are supposed to change them.

File: src/data_provider/src/packages/packageLinuxRpmParserHelperLegacy.h (hunk 5 above)
Comment (RamiroRD) [reply src/data_provider/src/packages/packageLinuxRpmParserHelperLegacy.h:4]: Same as above

**Review by HanesSciarrone — DISMISSED** [review]

**Review by RamiroRD — APPROVED** [review]
lgtm!

**Review by RamiroRD — APPROVED** [review]

**Review by HanesSciarrone — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**